### DISCUSSION TITLE
Problem parsing quoted string which may contain backslash followed by quote

### DISCUSSION THREAD

**Discussion by Bobbias** [foonathan/lexy#65] [Q&A] [answered]
I'm attempting to build a parser for the COOL language. (Not for any class, but just as a personal project)

To handle strings in cool there are not really any escape sequences. Sequences such as \t and \n are retained as part of the string literal. Strings may also contain doublequotes, provided they are preceded with a backslash:

`"This is an \"example string\" in cool."`

This string should be printed to the screen as:

`This is an \"example string\" in cool.`

Unlike escape sequences, the backslash is not consumed.

I simply cannot figure out how to approach this.

**Comment by foonathan** [comment, marked as answer]
The easiest way is to simply use a backslash escape sequence, where the value of `"` is `\"``: https://godbolt.org/z/197PoK54v

**Reply by Bobbias** [reply]
That was the first idea I had, but when the backslash simply wasn't being printed no matter what I did, I got quite frustrated and wondered if the escape functionality was doing something I didn't understand.

The actual problem I had was that I was still using `symbol_table<char>` with a multi-char character, and it appears that the character which was being used happened to be the double quote rather than the backslash. I'm guessing this comes down to implementation-specific details combined with the system being little-endian, since re-ordering things in compiler explorer shows that the second character is the one that gets printed.

I also didn't know the raw string syntax, so I was going through the trouble of escaping things, leading to quite a mess `'\\\"'`.

It's been a long time since I'm written much C++, and to make things worse, I mostly learned the language from a book that predated the c++98 standard.

I would have included code in my original post, but at the time I was tired and frustrated and just didn't have the energy. I really appreciate the quick reply.

(Also, where exactly is the button to get program output to display on compiler explorer? I opened another page to experiment, but could not seem to find a way to see the stdout.)

**Reply by foonathan** [reply]
You need to add an "executor" to compiler explorer.

**Reply by Bobbias** [reply]
I tried that, and while I could see the compilation output, I never saw my program's output.

I can work around it by opening any link which already has a working output, and reusing that page, but for whatever reason it seems to refuse to execute my code when starting fresh.

Edit: Figured things out. I was not reading carefully enough. My program was crashing.
